ibis/backends/mssql/compiler.py

```diff
@@ -1,39 +1,599 @@
 from __future__ import annotations
 
-from sqlalchemy.dialects.mssql import DATETIME2
+import calendar
+import operator
+from functools import reduce, singledispatchmethod
+from itertools import chain
 
-import ibis.expr.operations as ops
-from ibis.backends.base.sql.alchemy import AlchemyCompiler, AlchemyExprTranslator
-from ibis.backends.mssql.datatypes import MSSQLType
-from ibis.backends.mssql.registry import _timestamp_from_unix, operation_registry
-from ibis.expr.rewrites import rewrite_sample
+import sqlglot as sg
+import sqlglot.expressions as sge
+from public import public
+from sqlglot.dialects import TSQL
+from sqlglot.dialects.dialect import rename_func
 
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+from ibis.backends.base.sqlglot.compiler import NULL, STAR, SQLGlotCompiler, paren
+from ibis.backends.base.sqlglot.datatypes import MSSQLType
+from ibis.backends.base.sqlglot.rewrites import (
+    rewrite_first_to_first_value,
+    rewrite_last_to_last_value,
+)
+from ibis.common.deferred import var
+from ibis.common.patterns import replace
+from ibis.expr.rewrites import p, rewrite_sample
 
-class MsSqlExprTranslator(AlchemyExprTranslator):
-    _registry = operation_registry
-    _rewrites = AlchemyExprTranslator._rewrites.copy()
-    _bool_aggs_need_cast_to_int32 = True
+TSQL.Generator.TRANSFORMS |= {
+    sge.ApproxDistinct: rename_func("approx_count_distinct"),
+    sge.Stddev: rename_func("stdevp"),
+    sge.StddevPop: rename_func("stdevp"),
+    sge.StddevSamp: rename_func("stdev"),
+    sge.Variance: rename_func("var"),
+    sge.VariancePop: rename_func("varp"),
+    sge.Ceil: rename_func("ceiling"),
+    sge.Trim: lambda _, e: f"TRIM({e.this.sql('tsql')})",
+}
 
-    _timestamp_type = DATETIME2
-    _integer_to_timestamp = staticmethod(_timestamp_from_unix)
+x = var("x")
+y = var("y")
+start = var("start")
+end = var("end")
 
-    native_json_type = False
 
-    _forbids_frame_clause = AlchemyExprTranslator._forbids_frame_clause + (
-        ops.Lag,
-        ops.Lead,
+@replace(
+    p.WindowFunction(
+        p.RowNumber | p.NTile | p.Lag | p.Lead | p.PercentRank | p.CumeDist, y
     )
-    _require_order_by = AlchemyExprTranslator._require_order_by + (ops.Reduction,)
-    _dialect_name = "mssql"
+)
+def exclude_unsupported_window_frame_from_ops(_, y):
+    return _.copy(frame=y.copy(start=None, end=0))
+
+
+@replace(p.Not(x))
+def replace_not_with_ifelse(_, x):
+    assert _.dtype.is_boolean()
+    return ops.Cast(ops.IfElse(x, 0, 1), _.dtype)
+
+
+@replace(p.Not(x))
+def replace_not_with_equals(_, x):
+    return ops.Equals(x, 0)
```

Comment:
I would inline this tule since it is easy enough to spell in a single statement:

```py
p.Not(x) >> d.Equals(x, 0)
```